### DISCUSSION TITLE
Does the badgemaker npm package call the shields.io API

### DISCUSSION THREAD

**Discussion by Abhi347** [badges/shields#7901] [Q&A] [answered]
I want to use the shields.io API to generate badges, but I received a security concern about using an external API. I just found out about existence of the `badgemaker` npm package and as far as I could check in the source code the badges are generated locally without calling any external API.  
I just wanted to confirm whether this is the case, if not, I'll have to take approval of different teams which I want to avoid. Any idea there?

**Comment by calebcartwright** [comment, marked as answer]
No, it does not. The Shields.io service and it's underlying API/badge server are users of the `badge-maker` package, not the other way around.

You can see that dependency here: https://github.com/badges/shields/blob/master/package.json#L29, and as you already noted, review the relatively minimal code base of badge-maker to see that's the case https://github.com/badges/shields/tree/master/badge-maker

**Comment by chris48s** [comment]
yep -  badge-maker is completely standalone :)
